lib/Cron/SocialUpdateRegistration.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * @copyright 2017 Georg Ehrke <oc.list@georgehrke.com>
+ *
+ * @author Georg Ehrke <oc.list@georgehrke.com>
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Cron;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\IUser;
+use OCP\IConfig;
+use OCP\IUserManager;
+
+class SocialUpdateRegistration extends \OC\BackgroundJob\TimedJob {
+	private $appName;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	/** @var IJobList */
+	private $jobList;
+
+	/** @var IConfig */
+	private $config;
+
+	/**
+	 * RegisterSocialUpdate constructor.
+	 *
+	 * @param ITimeFactory $time
+	 * @param IUserManager $userManager
+	 * @param IJobList $jobList
+	 */
+	public function __construct(string $AppName,
+					//  ITimeFactory $time,
+					IUserManager $userManager,
+					IConfig $config,
+					IJobList $jobList) {
+		//parent::__construct($time);
+		
+		$this->appName = $AppName;
+		$this->userManager = $userManager;
+		$this->config = $config;
+		$this->jobList = $jobList;
+
+		// Run once a week
+		parent::setInterval(7 * 24 * 60 * 60);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	protected function run($arguments) {
+
+		// check if admin allows for social updates:
+		$isAdminEnabled = $this->config->getAppValue($this->appName, 'allowSocialSync', 'yes');
```

Comment:
I would go for a different variable name.
Something like `$isSyncEnabledByAdmin`

Reply:
Now Codacy complains:

> Avoid excessively long variable names like $isSocialEnabledByAdmin. Keep variable name length under 20.

Shall I go back to the previous variable names?

Reply:
xD, Ignore codacy, I don't even know why I enabled it in the first place :p

Reply:
I think I found a compromise :wink: